### PR TITLE
Fix set value on nested field

### LINKED ISSUE #56: Can't set value on embedded struct fields
`structs.Field.Set` fails when trying to set a value on an embedded struct field because [`CanSet`](https://github.com/fatih/structs/blob/master/field.go#L73) returns `false`.

It should not fail since the documentation of `CanSet` says:

> A Value can be changed only if it is addressable and was not obtained by the use of unexported struct fields

Here is an example that you can copy-paste:

```go
package my_test

import (
	"reflect"
	"testing"

	"github.com/fatih/structs"
)

type Base struct {
	ID int
}

type User struct {
	Base
	Name string
}

// This test fails
func TestEmbeddedWithStructs(t *testing.T) {
	u := User{}
	s := structs.New(&u)
	f := s.Field("Base").Field("ID")
	err := f.Set(10)
	if err != nil {
		t.Errorf("Error %v", err)
	}
	if f.Value().(int) != 10 {
		t.Errorf("Value should be equal to 10, got %v", f.Value())
	}
}

// This test works fine
func TestEmbeddedWithReflect(t *testing.T) {
	u := User{}
	s := reflect.ValueOf(&u).Elem()
	f := s.FieldByName("Base").FieldByName("ID")
	if !f.CanSet() {
		t.Error("CanSet should be true")
	}
	f.Set(reflect.ValueOf(10))
	if f.Interface().(int) != 10 {
		t.Errorf("Value should be equal to 10, got %v", f.Interface())
	}
}

```

### PR DESCRIPTION
**Pull request by asdine**
Fixes #56

### CODE REVIEW

File: field.go [hunk 1]
```diff
@@ -117,7 +117,12 @@ func (f *Field) Field(name string) *Field {
 // FieldOk returns the field from a nested struct. The boolean returns whether
 // the field was found (true) or not (false).
 func (f *Field) FieldOk(name string) (*Field, bool) {
-	v := strctVal(f.value.Interface())
+	value := &f.value
+	if f.value.Kind() != reflect.Ptr {
+		a := f.value.Addr()
+		value = &a
+	}
```
Comment (fatih) [inline field.go]: Mind adding a comment why you take an address here so in the future we know what's happened :)

**Comment by fatih** [comment]
Thanks @asdine for the fix. Just wrote a comment, can you PTAL ?

File: field.go (hunk 1 above)
Comment (asdine) [inline field.go]: Sure :)

**Comment by fatih** [comment]
Thanks @asdine 👍

**Pull request merged**